File: backend/internal/models/chanting.go
package models

import "time"

// Chanting represents a Buddhist chant with multi-language titles, dual Pali scripts, translations, and audio support.
type Chanting struct {
	ID              int           `gorm:"primaryKey;autoIncrement" json:"id"`
	Slug            string        `gorm:"size:120;uniqueIndex;not null" json:"slug"`
	Title           MultiLangText `gorm:"type:jsonb;not null" json:"title"`
	Subtitle        MultiLangText `gorm:"type:jsonb" json:"subtitle"`
	Category        string        `gorm:"size:50;not null;default:'general';index" json:"category"`
	PaliThai        string        `gorm:"type:text;not null" json:"pali_thai"`
	PaliRoman       string        `gorm:"type:text;not null" json:"pali_roman"`
	Translation     MultiLangText `gorm:"type:jsonb;not null" json:"translation"`
	AudioURL        string        `gorm:"size:255" json:"audio_url"`
	DurationSeconds int           `gorm:"default:0" json:"duration_seconds"`
	DisplayOrder    int           `gorm:"default:0;index" json:"display_order"`
	IsActive        bool          `gorm:"default:true;index" json:"is_active"`
	CreatedAt       time.Time     `json:"created_at"`
	UpdatedAt       time.Time     `json:"updated_at"`
}

File: frontend/src/types/chanting.ts
import type { MultiLangText } from "./api";

export type ChantingCategory =
  | "all"
  | "general"
  | "morning_chant"
  | "evening_chant"
  | "paritta"
  | "blessing"
  | "funeral";

export interface Chanting {
  id: number;
  slug: string;
  title: MultiLangText;
  subtitle?: MultiLangText;
  category: ChantingCategory;
  pali_thai: string;
  pali_roman: string;
  translation: MultiLangText;
  audio_url?: string;
  duration_seconds: number;
  display_order: number;
  is_active: boolean;
  created_at?: string;
  updated_at?: string;
}

export type PaliScript = "thai" | "roman";
